cloud/blockstore/libs/storage/volume/volume_actor_forward_trackused.cpp

```diff
@@ -36,23 +36,25 @@ bool TVolumeActor::SendRequestToPartitionWithUsedBlockTracking(
         State->IsDiskRegistryMediaKind() &&
         volumeConfig.GetEncryptionDesc().GetMode() != NProto::NO_ENCRYPTION;
     const bool overlayDiskRegistryBasedDisk =
-        State->IsDiskRegistryMediaKind() &&
-        !State->GetBaseDiskId().Empty();
+        State->IsDiskRegistryMediaKind() && !State->GetBaseDiskId().Empty();
 
     if constexpr (IsWriteMethod<TMethod>) {
-        if (State->GetTrackUsedBlocks() ||
-            State->HasCheckpointLight() ||
-            overlayDiskRegistryBasedDisk)
-        {
+        if (State->GetTrackUsedBlocks() || State->HasCheckpointLight()) {
             auto requestInfo =
                 CreateRequestInfo(ev->Sender, ev->Cookie, msg->CallContext);
 
+            // For overlay disks, we must ensure that the map of used blocks
```

Comment:
For overlay disks we need to ensure that the only bits that are set in the used block map are the bits that correspond to the blocks that have been successfully written. Therefore we must update the map only after the block has been successfully written.